src/domain/settings/currencies/components/currency-tax-setting.tsx

```diff
@@ -0,0 +1,85 @@
+import { Currency } from "@medusajs/medusa"
+import { useAdminUpdateCurrency } from "medusa-react"
+import React, { useEffect } from "react"
+import { Controller, useForm } from "react-hook-form"
+import Switch from "../../../../components/atoms/switch"
+import CoinsIcon from "../../../../components/fundamentals/icons/coins-icon"
+import useNotification from "../../../../hooks/use-notification"
+import { getErrorMessage } from "../../../../utils/error-messages"
+
+type CurrencyTaxSettingFormType = {
+  includes_tax: boolean
+}
+
+type Props = {
+  currency: Currency
+  isDefault: boolean
+}
+
+const CurrencyTaxSetting = ({ currency, isDefault }: Props) => {
+  const { mutate } = useAdminUpdateCurrency(currency.code)
+  const { handleSubmit, control, reset } = useForm<CurrencyTaxSettingFormType>({
+    defaultValues: {
+      includes_tax: currency.includes_tax,
+    },
+  })
+
+  const notification = useNotification()
+
+  useEffect(() => {
+    reset({
+      includes_tax: currency.includes_tax,
+    })
+  }, [currency])
+
+  const onSubmit = handleSubmit((data: CurrencyTaxSettingFormType) => {
+    mutate(data, {
+      onSuccess: () => {
+        notification("Success", "Successfully updated currency", "success")
+      },
+      onError: (error) => {
```

Comment:
suggestion: using `mutateAsync` will properly catch error, this way if we have 4xx the onError won't fire

Reply:
Why would it not catch the errors as is? Just tried it, and catches 400s without issue, is this not what you are experiencing?

Reply:
Hmm, I get an uncaught error if it's 404

Reply:
Okay super interesting, can you give me some steps to try and reproduce? If I ensure a 404, it catches it properly and gives the correct error message in a notification.

![image](https://user-images.githubusercontent.com/45367945/190394846-bf6f5b4d-51f9-4e26-b21c-dadf6898fb44.png)

Reply:
Okay, did some digging and it seems like `medusa-react` can't handle 404's that a thrown as a result of a route not existing, but handles "actual" 404's perfectly fine.